pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -465,7 +465,9 @@ func (c *Cluster) EnableServiceMonitor(activeDaemon string) error {
 	}
 	serviceMonitor.SetName(AppName)
 	serviceMonitor.SetNamespace(c.clusterInfo.Namespace)
+
 	cephv1.GetMonitoringLabels(c.spec.Labels).ApplyToObjectMeta(&serviceMonitor.ObjectMeta)
```

Comment:
`ApplyToObjectMeta()` will overwrite labels with new values if already specified. So how about the pattern of initializing the labels to the default labels, then call `ApplyToObjectMeta()`? This should have the desired effect, and no need to implement the `applyDefaultMonitoringLabels()` method.
```suggestion
	serviceMonitor.Labels = defaultLabels
	cephv1.GetMonitoringLabels(c.spec.Labels).ApplyToObjectMeta(&serviceMonitor.ObjectMeta)
```

Reply:
Jumping back into thread (Fasthub App couldn't access this repo and E-Mail didn't land in thread)

Setting the default labels in code and then calling ApplyToObjectMeta would result in the same behavior as I'm trying to fix, as it will not override labels that are already set. Previously they were loaded from the template files, setting them in go code before that call is basically the same as before.

`// ApplyToObjectMeta adds labels to object meta unless the keys are already defined.`
https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/labels.go#L79

Reply:
Oh you're right, not sure what I was thinking yesterday... 

I'm wondering if we could just simplify everything by changing `ApplyToObjectMeta()` to overwrite existing labels instead of skipping updates to existing labels. Then I believe the scenario would be covered with just a 1-2 line update to `ApplyToObjectMeta()`, and no other changes (except unit tests).  I'll investigate and report back later today on that question...

Reply:
Changing the `ApplyToObjectMeta()` to always overwrite the labels may be fine in most cases, but there are a few places Rook relies on the labels it sets internally. For example, labels on OSDs when backed by PVCs, the operator needs to query specific labels to find the PVCs. So in the general case we better not allow labels to be overridden everywhere. 

But for monitoring, I don't see any reason not to overwrite the existing labels. What if you create a new method for overwriting labels? Everywhere we allow overrides from the user it would look something like this:
```
cephv1.GetMonitoringLabels(c.spec.Labels).ApplyToObjectMetaAllowOverrides(&serviceMonitor.ObjectMeta)
```

Then there is still no need to remove defaults from the yaml, right? You could just override the defaults.

Reply:
@galexrt Thoughts on overriding the labels?

Reply:
Having another method allowing to overwrite labels is a nice idea, I added that as `OverwriteApplyToObjectMeta` (thinking of the operation to be `overwrite-apply`, hence the different name). This also does not require changing the templates, so by default we have all the labels as before.